python/cudf/cudf/core/index.py

```diff
@@ -857,6 +881,43 @@ def min(self):
     def max(self):
         return self._minmax("max")
 
+    @property  # type: ignore
+    def values(self):
+        return cupy.arange(self.start, self.stop, self.step)
+
+    def to_frame(self, index=True, name=None):
+        """Create a DataFrame with a column containing this Index"""
+
+        if name is not None:
+            col_name = name
+        elif self.name is None:
+            col_name = 0
+        else:
+            col_name = self.name
+        return cudf.DataFrame(
+            {col_name: self._range}, index=self if index else None
+        )
+
+    def isin(self, values):
+        """Return a boolean array where the index values are in values
+
+        Compute boolean array of whether each index value is found in
+        the passed set of values. The length of the returned boolean
+        array matches the length of the index.
+        """
+
+        # To match pandas behavior, even though only list-like objects are
+        # supposed to be passed, only scalars throw errors. Other types (like
+        # dicts) just transparently return False (see the implementation of
+        # ColumnBase.isin).
+        if is_scalar(values):
+            raise TypeError(
+                "only list-like objects are allowed to be passed "
+                f"to isin(), you passed a {type(values).__name__}"
+            )
+
+        return np.isin(self.values_host, values)
```

Comment:
This is an interesting one. Can you run a few benchmarks to see how the performance of this implementation compares to a cupy implementation (`cp.isin(self.values, values)`) for different data sizes? The cupy implementation will require copying more data to the GPU, but then it should parallelize the check, so the latter should be faster for large data but slower for small data. I'm curious whether the tradeoff is and whether it makes sense to implement it that way.

Reply:
Here are the benchmark comparison-results for 001 (numpy) vs. 002 (cupy).  As shown, numpy implem. is much faster for both [100, 1000] rows, cupy gives very little to no advantage for larger (1000k rows), so maybe we should stick to numpy? what are your thoughts @vyasr ?
```
------------------------------- benchmark '100': 2 tests ------------------------------
Name (time in us)               Min                   Max                Mean          
---------------------------------------------------------------------------------------
bench_isin[100] (0001)      31.7814 (1.0)         87.9364 (1.0)       34.1955 (1.0)    
bench_isin[100] (0002)     648.8832 (20.42)    1,082.2723 (12.31)    754.7593 (22.07)  
---------------------------------------------------------------------------------------

------------------------------- benchmark '10000': 2 tests ------------------------------
Name (time in us)                 Min                   Max                Mean          
-----------------------------------------------------------------------------------------
bench_isin[10000] (0001)      42.1545 (1.0)      1,361.9522 (1.0)       44.7062 (1.0)    
bench_isin[10000] (0002)     630.6740 (14.96)    4,239.8935 (3.11)     657.1183 (14.70)  
-----------------------------------------------------------------------------------------

--------------------------- benchmark '1000000': 2 tests --------------------------
Name (time in ms)                 Min               Max              Mean          
-----------------------------------------------------------------------------------
bench_isin[1000000] (0001)     1.2690 (1.08)     1.4115 (1.0)      1.2835 (1.02)   
bench_isin[1000000] (0002)     1.1752 (1.0)      3.7461 (2.65)     1.2583 (1.0)    
-----------------------------------------------------------------------------------

Legend:
  Outliers: 1 Standard Deviation from Mean; 1.5 IQR (InterQuartile Range) from 1st Quartile and 3rd Quartile.
  OPS: Operations Per Second, computed as 1 / Mean
```

Reply:
Yeah I think we probably stick to numpy here. I guess if we really wanted to be fancy we could dispatch depending on the sizes of `self` and `values`, but IMO that's unnecessary.